GamesApp/GamesApp/Scenes/Dice/DiceViewController.swift

```diff
@@ -167,52 +193,23 @@ private extension GameViewController {
         
     }
     
-    func GameButton(){
+    func diceButtonAction() {
+        let request = DiceModels.States.Request(buttonState: playButton.isSelected)
+        interactor?.playTheGame(request: request)
         playButton.isSelected = !playButton.isSelected
-        if !playButton.isSelected {
-            rollTheDice()
-        }
-        else {
-            renewGame()
-        }
     }
-    
-    func rollTheDice(){
-        guard let chosenLabel = labels.randomElement() else {return}
-        guard let result = chosenLabel.text else {return}
-        diceDelegate?.getDiceHistory(diceData: result)
-        allDiceCounter += 1
-        switch chosenLabel {
-        case oneDotLabel:
-            oneCounter += 1
-        case twoDotLabel:
-            twoCounter += 1
-        case threeDotLabel:
-            threeCounter += 1
-        case fourDotLabel:
-            fourCounter += 1
-        case fiveDotLabel:
-            fiveCounter += 1
-        case sixDotLabel:
-            sixCounter += 1
-        default:
-            break
-        }
-        diceDelegate?.getDiceStats(
-            one: oneCounter/allDiceCounter,
-            two: twoCounter/allDiceCounter,
-            three: threeCounter/allDiceCounter,
-            four: fourCounter/allDiceCounter,
-            five: fiveCounter/allDiceCounter,
-            six: sixCounter/allDiceCounter)
-        
-        labels.forEach { $0.isHidden = true }
-        chosenLabel.isHidden = false
-        playButton.setTitle("Повтор", for: .normal)
+}
+
+extension DiceViewController: DiceDisplayLogic {
+    func getTheDice(viewModel: DiceModels.States.ViewModel.Play) {
+        playButton.setTitle(viewModel.buttonName, for: .normal)
+        labels.forEach { $0.isHidden = true}
+        labels[viewModel.shownLabel.rawValue].isHidden = false
+        router?.sendDiceResult()
     }
     
-    func renewGame(){
-        labels.forEach { $0.isHidden = false }
-        playButton.setTitle("Бросить кубик", for: .normal)
+    func revertDice(viewModel: DiceModels.States.ViewModel) {
+        playButton.setTitle(viewModel.buttonName, for: .normal)
```

Comment:
Dont worry to set button's title directly in ViewController avoiding getting title from VIP cycle.